### ISSUE TITLE
Remove temporary commits 

### ISSUE THREAD

**Issue by sechkova** [theupdateframework/tuf#1320]
**Description of issue or feature request**:

Remove commits that:

- temporary reduce code coverage percentage: 5c58c1e
- temporary disable linter checks: d252818, 062bda4

**Comment by jku** [comment]
pylint disables that are left after #1396:
* `consider-using-with` in download.py: I think this is correct, we want to return the tempfile so cannot use a context manager. Unless there is a way to write the function explicitly as a context manager itself (since that's how we use it anyway)?
* `broad-except` in metadata_bundle.py: This is in code that should be implemented in Metadata API as part of #1377 
* `too-many-branches` in metadata_bundle.py: I don't want to split update_snapshot(), also this should go away with #1361

I'm fine with those.

On the coverage I suggest we change tox to do 
`python -m coverage report -m --fail-under 97 --omit "{toxinidir}/tuf/client_rework/*"
`
for now, and file issues for improving this sad result:
```
/home/jku/src/tuf/tuf/client_rework/download.py
113->114, 114-119, 215->216, 216-243, 252->253, 253
/home/jku/src/tuf/tuf/client_rework/metadata_bundle.py
94->101, 95->103, 99->exit, 101, 103->104, 104, 115-117, 157, 161, 198->199, 199, 209->210, 210, 214->216, 216-222, 243->244, 244, 282->283, 283, 289->294, 294, 300->301, 301, 320->321, 321, 338-339, 341->342, 342, 346->347, 347, 351->355, 355, 361->362, 362-373, 378->379, 379, 421->422, 422, 427->428, 428-433, 451->452, 452, 457->458, 458
/home/jku/src/tuf/tuf/client_rework/requests_fetcher.py
102->103, 103, 136-137, 152->153, 153
/home/jku/src/tuf/tuf/client_rework/updater_rework.py
65->66, 66, 74->77, 77, 147->148, 148, 167->168, 168-170, 195->196, 196, 200->203, 203, 253->268, 258-259, 262->263, 263, 299->300, 300, 319->320, 320, 353->354, 354-356, 375->378, 388->389, 389-398, 400->401, 401-402, 421->426, 426-430, 479->480, 480-485, 487->523, 489->529, 499->511, 511-529, 540->541, 541, 556->557, 557, 571-576
```
EDIT: there is an umbrella issue #1309

**Comment by sechkova** [comment]
> On the coverage I suggest we change tox to do
> `python -m coverage report -m --fail-under 97 --omit "{toxinidir}/tuf/client_rework/*" `

This is done with 18c527241cdd8267bb3507bd73ec734ac0fed1a4.
Closing this issue and the remaining work will continue in #1309.
